### PR TITLE
*: add comment for exported elements

### PR DESCRIPTION
**Pull request by lonng**
Signed-off-by: Lonng <chris@lonng.org>

### What problem does this PR solve? 

Add comment for exported elements to make linter happy.

### What is changed and how it works?

### CODE REVIEW

**Comment by lonng** [comment]
PTAL @kennytm @amyangfei

File: code/rewriter.go [hunk 1]
```diff
@@ -512,6 +516,14 @@ func (r *Rewriter) rewriteFile(path string) (err error) {
 	return printer.Fprint(newFile, fset, file)
 }
 
+// Rewrite does the rewrite action for specified path. It is contains the main steps:
```
Comment (amyangfei) [inline code/rewriter.go]:
```suggestion
// Rewrite does the rewrite action for specified path. It contains the main steps:
```

File: code/rewriter.go [hunk 2]
```diff
@@ -512,6 +516,14 @@ func (r *Rewriter) rewriteFile(path string) (err error) {
...
+//
+// 1. Filter out files if it has not suffix `.go` or it is failpoint binding file
```
Comment (amyangfei) [inline code/rewriter.go]:
```suggestion
// 1. Filter out failpoint bind files and files that have no suffix `.go`
```

File: failpoint.go [hunk 3]
```diff
@@ -35,10 +44,14 @@ func (fp *failpoint) Pause() {
 	<-fp.waitChan
 }
 
+// WithHook binding a hook to a new context which base on the `ctx` parameter
```
Comment (amyangfei) [inline failpoint.go]:
```suggestion
// WithHook binds a hook to a new context which is based on the `ctx` parameter
```

File: code/rewriter.go [hunk 4]
```diff
@@ -33,6 +33,9 @@ const (
 	extendPkgName   = "_curpkg_"
 )
 
+// Rewriter represents a rewriting tool for converting the failpoint markers function to
+// corresponding statements in Golang. It will traverse the specified path and filter
+// out files which does not have failpoint injection sites, and rewrite the remain files.
```
Comment (amyangfei) [inline code/rewriter.go]:
```suggestion
// out files which do not have failpoint injection sites, and rewrite the remain files.
```

File: code/rewriter.go [hunk 5]
```diff
@@ -512,6 +516,14 @@ func (r *Rewriter) rewriteFile(path string) (err error) {
 	return printer.Fprint(newFile, fset, file)
 }
 
+// Rewrite does the rewrite action for specified path. It contains the main steps:
+//
+// 1. Filter out failpoint bind files and files that have no suffix `.go`
```
Comment (amyangfei) [inline code/rewriter.go]:
```suggestion
// 1. Filter out failpoint binding files and files that have no suffix `.go`
```

**Comment by amyangfei** [comment]
Rest LGTM

File: code/rewriter.go [hunk 6]
```diff
@@ -33,6 +33,9 @@ const (
 	extendPkgName   = "_curpkg_"
 )
 
+// Rewriter represents a rewriting tool for converting the failpoint markers function to
```
Comment (kennytm) [inline code/rewriter.go]:
```suggestion
// Rewriter represents a rewriting tool for converting the failpoint marker functions to
```

File: code/rewriter.go [hunk 7]
```diff
@@ -512,6 +516,14 @@ func (r *Rewriter) rewriteFile(path string) (err error) {
 	return printer.Fprint(newFile, fset, file)
 }
 
+// Rewrite does the rewrite action for specified path. It contains the main steps:
+//
+// 1. Filter out failpoint bind files and files that have no suffix `.go`
+// 2. Filter out files which have not import failpoint package (implies no failpoints)
```
Comment (kennytm) [inline code/rewriter.go]:
```suggestion
// 2. Filter out files which have not imported failpoint package (implying no failpoints)
```

File: code/rewriter.go [hunk 8]
```diff
@@ -512,6 +516,14 @@ func (r *Rewriter) rewriteFile(path string) (err error) {
...
+// 3. Parse file to `ast.File` and rewrite the AST
+// 4. Create failpoint binding file (which contains `_curpkg_` function) if it not exists
```
Comment (kennytm) [inline code/rewriter.go]:
```suggestion
// 4. Create failpoint binding file (which contains `_curpkg_` function) if it does not exist
```

File: failpoint.go [hunk 9]
```diff
@@ -35,10 +44,14 @@ func (fp *failpoint) Pause() {
 	<-fp.waitChan
 }
 
+// WithHook binds a hook to a new context which is based on the `ctx` parameter
 func WithHook(ctx context.Context, hook Hook) context.Context {
 	return context.WithValue(ctx, failpointCtxKey, hook)
 }
 
+// EvalContext evaluates a failpoint's value, will call hook if the context is
```
Comment (kennytm) [inline failpoint.go]:
```suggestion
// EvalContext evaluates a failpoint's value, and calls hook if the context is
```

**Pull request merged**